lib/internal/test_runner/yaml_to_js.js

```diff
@@ -44,9 +44,11 @@ function reConstructError(parsedYaml) {
     cause = new Error(parsedYaml.error);
     cause.code = parsedYaml.code;
   }
-  cause.stack = stack;
+  const name = parsedYaml.name ?? 'Error';
+  cause.stack = `${name}: ${parsedYaml.error}\n${stack}`;
 
   if (isTestFailure) {
+    if (!isAssertionError) delete cause.code;
```

Comment:
Instead of deleting the property after setting it, maybe not set it, if it's a test failure?

https://github.com/nodejs/node/pull/46872/files#diff-d684728ae3b692ada077eb5c9f2b5d0f8d1462916b1675b4624914496c8bd1daR45